File: project/backend/src/main/java/com/hairapy/models/Salon.java
package com.hairapy.models;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

// Salon doi tac hien thi o SalonsPage. Truoc day FE dung mock data hardcode trong figmaAssets.js.
// service_types luu comma-separated giong pattern face_shape cua HairstyleCatalog.
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "salons", indexes = {
        @Index(name = "idx_salons_district", columnList = "district")
})
public class Salon {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    @Column(nullable = false, length = 150)
    private String name;

    @NotBlank
    @Column(nullable = false, length = 300)
    private String address;

    // "Quận 1", "Quận 3", "Bình Thạnh"...
    @NotBlank
    @Column(nullable = false, length = 50)
    private String district;

    // Comma-separated: "Cắt tóc,Nhuộm tóc,Combo trọn gói"
    @Column(name = "service_types", length = 300)
    private String serviceTypes;

    // Giá khởi điểm (VND)
    @Column(name = "price_from", nullable = false)
    private Integer priceFrom;

    @Column(nullable = false)
    @Builder.Default
    private Double rating = 0.0;

    @Column(nullable = false)
    @Builder.Default
    private boolean verified = false;

    @NotBlank
    @Column(name = "image_url", nullable = false, length = 500)
    private String imageUrl;

    @Column(length = 20)
    private String phone;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;
}

File: project/mobile/lib/models/salon.dart
/// Khớp entity Salon — GET /salons (danh sách salon đối tác, công khai).
class Salon {
  final int id;
  final String name;
  final String address;
  final String district;
  final String? serviceTypes; // comma-separated: "Cắt tóc,Nhuộm tóc"
  final int priceFrom;
  final double rating;
  final bool verified;
  final String imageUrl;
  final String? phone;

  Salon({
    required this.id,
    required this.name,
    required this.address,
    required this.district,
    required this.priceFrom,
    required this.rating,
    required this.verified,
    required this.imageUrl,
    this.serviceTypes,
    this.phone,
  });

  List<String> get services =>
      (serviceTypes ?? '').split(',').map((e) => e.trim()).where((e) => e.isNotEmpty).toList();

  factory Salon.fromJson(Map<String, dynamic> json) => Salon(
        id: json['id'] as int,
        name: json['name'] as String? ?? '',
        address: json['address'] as String? ?? '',
        district: json['district'] as String? ?? '',
        serviceTypes: json['serviceTypes'] as String?,
        priceFrom: json['priceFrom'] as int? ?? 0,
        rating: (json['rating'] as num?)?.toDouble() ?? 0,
        verified: json['verified'] as bool? ?? false,
        imageUrl: json['imageUrl'] as String? ?? '',
        phone: json['phone'] as String?,
      );
}
